toto/models/matchrule.py

```diff
@@ -0,0 +1,102 @@
+"""
+<Module Name>
+  toto/models/validators.py
+
+<Author>
+  Santiago Torres-Arias <santiago@nyu.edu>
+
+<Started>
+  Nov 18, 2016
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  This module provides functions to ensure the syntax of the matchrules is
+  correct.
+
+"""
+from toto.ssl_commons.exceptions import FormatError
+import toto.ssl_crypto.formats
+
+def _validate_match_rule(keywords):
+  """ private helper to verify the syntax of the MATCH matchrule """
+  MATERIAL_OR_PRODUCT = {'PRODUCT', 'MATERIAL'}
+
+  if not isinstance(keywords, list):
+    raise FormatError("this matching rule is not a list")
+
+  if len(keywords) == 5:
+    rule, artifact, target, from_keyword, step = keywords
+    as_keyword = "AS"
+    new_target = target
+
+  elif len(keywords) == 7:
+    rule, artifact, target, from_keyword, step, as_keyword, new_target = keywords
```

Comment:
Please take a look at the current [matchrule implementation](https://github.com/in-toto/in-toto/blob/develop/toto/verifylib.py#L280-L285)
It's actually:
```
MATCH (MATERIAL|PRODUCT) <path pattern>  [AS <target path pattern>] FROM <step name>
```